corehq/apps/api/resources/v0_5.py

```diff
@@ -268,15 +268,14 @@ def obj_create(self, bundle, **kwargs):
             bundle.obj.save()
         except Exception:
             if bundle.obj._id:
-                bundle.obj.retire(deleted_by=bundle.request.user, deleted_via=USER_CHANGE_VIA_API)
+                bundle.obj.retire(bundle.request.domain, deleted_by=bundle.request.couch_user,
+                                  deleted_via=USER_CHANGE_VIA_API)
             try:
                 django_user = bundle.obj.get_django_user()
             except User.DoesNotExist:
                 pass
             else:
                 django_user.delete()
-                log_model_change(bundle.request.user, django_user, message=f"deleted_via: {USER_CHANGE_VIA_API}",
-                                 action=ModelAction.DELETE)
```

Comment:
since the logging is already happening on retire, this additional logging should not be needed.

Reply:
`retire` only gets called if `bundle.obj._id` evaluates to true, but this could get called even if that is false, right? Maybe we want to add that condition to this logging? Otherwise there may be changes that are not logged.

Reply:
Hey @millerdev 

You are right that if the django user object is deleted without couch user, there won't be a log for it.
There is a bit more happening here. Basically, it's simply a clean up of the django object if the couch user fails to create. If the couch user was never created, logging this isn't of use. Plus the log needs the couch user ID.
I believe the original PR that added this could have skipped it as well.

Let me know if that makes sense.